roofit/roofit/CMakeLists.txt

```diff
@@ -146,6 +148,39 @@ ROOT_STANDARD_LIBRARY_PACKAGE(RooFit
     MathCore
 )
 
+# RooFitComputeLib.cxx is already included in the RooFit package once, as a default architecture generic implementation.
+# If roofit-arch-libs is enabled (see CMakeLists.txt in the parent directory) , we want to compile the file again as a seperate library.
+# For each copy we give the corresponding flags and at runtime, when cpu detection takes place the correct library is loaded dynamically.
+if (roofit-arch-libs)
+  ROOT_LINKER_LIBRARY(RooFitCompute_SSE4.1  src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
+  ROOT_LINKER_LIBRARY(RooFitCompute_AVX     src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
+  ROOT_LINKER_LIBRARY(RooFitCompute_AVX2    src/RooFitComputeLib.cxx TYPE SHARED DEPENDENCIES RooFit)
+  
+  # Add -fno-signaling-nans, -fno-trapping-math and -O3 to enable autovectorization
+  # of the RooFitComputeLibrary (especially needed for GCC).
+  set(common-flags $<$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebInfo>>:$<$<CXX_COMPILER_ID:GNU>:-fno-signaling-nans> -fno-trapping-math -O3>)
+  target_compile_options(RooFitCompute_SSE4.1  PRIVATE ${common-flags} -msse4    -DRF_ARCH=SSE4)
+  target_compile_options(RooFitCompute_AVX     PRIVATE ${common-flags} -mavx     -DRF_ARCH=AVX )
+  target_compile_options(RooFitCompute_AVX2    PRIVATE ${common-flags} -mavx2    -DRF_ARCH=AVX2)
+
+  if(vdt AND NOT builtin_vdt)
```

Comment:
What happens if this `if` were written as `if (vdt)`?

Reply:
I don't know. I found it like this, some lines after... I guess it will not find the custom directory or maybe it will and the command is reduntant.